pyEpiabm/pyEpiabm/tests/test_unit/test_routine/test_simulation.py

```diff
@@ -483,6 +484,77 @@ def test_write_to_ih_file_status_infectiousness(self, mock_mkdir, time=1):
         mock_mkdir.assert_called_with(
             os.path.join(os.getcwd(), self.inf_history_params["output_dir"]))
 
+    @patch('os.makedirs')
+    def test_compress_no_compression(self, mock_mkdir):
+        mo = mock_open()
+        self.inf_history_params['compress'] = False
+        self.inf_history_params['status_output'] = True
+        self.inf_history_params['infectiousness_output'] = True
+
+        with patch('pyEpiabm.output._csv_dict_writer.open', mo):
+            test_sim = pe.routine.Simulation()
+            mock_status_writer = MagicMock()
+            mock_infectiousness_writer = MagicMock()
+            test_sim.configure(self.test_population, self.initial_sweeps,
+                               self.sweeps, self.sim_params, self.file_params,
+                               self.inf_history_params)
+            test_sim.run_sweeps()
+
+            with patch.object(test_sim, 'ih_status_writer',
+                              mock_status_writer):
+                with patch.object(test_sim, 'ih_infectiousness_writer',
+                                  mock_infectiousness_writer):
+                    test_sim.compress_csv()
+                    mock_status_writer.compress.assert_not_called()
+                    mock_infectiousness_writer.compress.assert_not_called()
+            del test_sim.writer
+            del test_sim.ih_status_writer
+            del test_sim.ih_infectiousness_writer
+
+    @patch('os.makedirs')
+    def test_compress_csv_status(self, mock_mkdir):
+        mo = mock_open()
+        self.inf_history_params['compress'] = True
+        self.inf_history_params['status_output'] = True
+        self.inf_history_params['infectiousness_output'] = False
+
+        with patch('pyEpiabm.output._csv_dict_writer.open', mo):
+            test_sim = pe.routine.Simulation()
+            mock_status_writer = MagicMock()
+            test_sim.configure(self.test_population, self.initial_sweeps,
+                               self.sweeps, self.sim_params, self.file_params,
+                               self.inf_history_params)
+            test_sim.run_sweeps()
+            with patch.object(test_sim, 'ih_status_writer',
+                              mock_status_writer):
+                test_sim.compress_csv()
+                mock_status_writer.compress.assert_called_once()
```

Comment:
Would it be possible here to test that it was called with the right data? If this is not easy then don't worry about it